tools/mpi-imm.cc

```diff
@@ -134,13 +134,39 @@ int main(int argc, char *argv[]) {
   weightGen.split(2, 0);
 
   using edge_type = ripples::WeightedDestination<uint32_t, float>;
+  #if defined ENABLE_METALL
+  using GraphFwd =
+      ripples::Graph<uint32_t, edge_type, ripples::ForwardDirection<uint32_t>, metall::manager::allocator_type<char>>;
+  using GraphBwd =
+      ripples::Graph<uint32_t, edge_type, ripples::BackwardDirection<uint32_t>, metall::manager::allocator_type<char>>;
+  #else
   using GraphFwd =
       ripples::Graph<uint32_t, edge_type, ripples::ForwardDirection<uint32_t>>;
   using GraphBwd =
       ripples::Graph<uint32_t, edge_type, ripples::BackwardDirection<uint32_t>>;
+  #endif
   console->info("Loading...");
-  GraphFwd Gf = ripples::loadGraph<GraphFwd>(CFG, weightGen);
+  #if defined ENABLE_METALL
+  bool exists = metall::manager::consistent(CFG.metall_dir.c_str());
+  metall::manager manager =
+      (exists ? metall::manager(metall::open_only, CFG.metall_dir.c_str())
+              : metall::manager(metall::create_only, CFG.metall_dir.c_str()));
+  GraphBwd *Gr;
+  if (exists) {
+    console->info("Previously existing graph exists! Loading...");
+    Gr = manager.find<GraphBwd>("graph").first;
+  } else {
+    console->info("Creating new metall directory...");
+    GraphFwd Gf =
+        ripples::loadGraph<GraphFwd>(CFG, weightGen, manager.get_allocator());
+    Gr = manager.construct<GraphBwd>("graph")(Gf.get_transpose());
+  }
+  GraphBwd &G(*Gr);
+#else
+  std::allocator<char> = GraphAllocator;
```

Comment:
It seems ripple build with MPI enabled would fail on this line due to this change. MPI disabled will build successfully. Error message:

```
../../tools/mpi-imm.cc: In function ‘int main(int, char**)‘:
../../tools/mpi-imm.cc:166:24: error: expected unqualified-id before ‘=’ token
 166 |  std::allocator<char> = GraphAllocator;
   |            ^
../../tools/mpi-imm.cc:167:62: error: ‘GraphAllocator’ was not declared in this scope
 167 |  GraphFwd Gf = ripples::loadGraph<GraphFwd>(CFG, weightGen, GraphAllocator);
   |                               ^~~~~~~~~~~~~~
Waf: Leaving directory `/global/u1/h/hxu398/ripples/build/release’
Build failed
 -> task in ‘mpi-imm’ failed with exit status 1 (run with -v to display more information)
```

Reply:
Hello, can you see if #8 fixes it? Just tested on my machine and it seems to compile now.

Reply:
I can confirm that #8 also fix the build on Perlmutter. Thank you for the prompt response!